src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -509,16 +509,18 @@ private function getShell()
 
     private function isTty(): bool
     {
-        if (!\defined('STDIN')) {
+        $inputStream = $this->inputStream ?: fopen('php://stdin', 'r');
+
+        if ('php://stdin' !== (stream_get_meta_data($inputStream)['url'] ?? null)) {
             return true;
         }
 
         if (\function_exists('stream_isatty')) {
-            return stream_isatty(fopen('php://input', 'r'));
+            return stream_isatty($inputStream);
```

Comment:
this was my mistake: using `php://stdin` here instead of `php://input` fixes the failure